.travis.sh

```diff
@@ -60,26 +60,58 @@ tests_sequence() {
     )
 }
 
+tests_sequence_unstable_target() {
+    # There's something odd with using the .cargo/config runner attribute and
+    # workspaces when the runner uses `cargo run --manifest-path ../Cargo.toml
+    # --bin cargo-dinghy ...`
+    title "testing from project directory for rust target $1"
+    ( \
+        cd test-ws/test-bin \
+        && cargo clean \
+        && cargo +nightly run -Zbuild-std --target $1 \
+    )
+}
+
 
 if [ `uname` = Darwin ]
 then
      title "••••• Darwin: ios simulator tests •••••"
      title "boot a simulator"
      rustup target add x86_64-apple-ios;
      RUNTIME_ID=$(xcrun simctl list runtimes | grep iOS | cut -d ' ' -f 7 | tail -1)
-     export SIM_ID=$(xcrun simctl create My-iphone7 com.apple.CoreSimulator.SimDeviceType.iPhone-8 $RUNTIME_ID)
+     export SIM_ID=$(xcrun simctl create My-iphone8 com.apple.CoreSimulator.SimDeviceType.iPhone-8 $RUNTIME_ID)
      xcrun simctl boot $SIM_ID
      tests_sequence $SIM_ID
 
      xcrun simctl delete $SIM_ID
-    
+
     if ios-deploy -c -t 1 > /tmp/ios_devices
     then
         device=$(grep "Found" /tmp/ios_devices | head -1 | cut -d " " -f 3)
         title "••••• Darwin: ios-deploy detected a device •••••"
         rustup target add aarch64-apple-ios
         tests_sequence $device
     fi
+
+     title "••••• Darwin: tvos simulator tests •••••"
+     title "boot a simulator"
+     rustup toolchain add nightly --component rust-src;
+     TVOS_RUNTIME_ID=$(xcrun simctl list runtimes | grep tvOS | cut -d ' ' -f 7 | tail -1)
+     export TV_SIM_ID=$(xcrun simctl create My-4ktv com.apple.CoreSimulator.SimDeviceType.Apple-TV-4K-3rd-generation-4K $TVOS_RUNTIME_ID)
+
+     xcrun simctl boot $TV_SIM_ID
+     tests_sequence_unstable_target x86_64-apple-tvos
+     xcrun simctl delete $TV_SIM_ID
+
+     title "••••• Darwin: watchvos simulator tests •••••"
+     title "boot a simulator"
+     rustup toolchain add nightly --component rust-src;
+     WATCHOS_RUNTIME_ID=$(xcrun simctl list runtimes | grep watchOS | cut -d ' ' -f 7 | tail -1)
+     export WATCHOS_SIM_ID=$(xcrun simctl create My-apple-watch com.apple.CoreSimulator.SimDeviceType.Apple-Watch-SE-44mm-2nd-generation $WATCHOS_RUNTIME_ID)
+
+     xcrun simctl boot $WATCHOS_SIM_ID
+     tests_sequence_unstable_target x86_64-apple-watchos-sim
```

Comment:
Similar to the M1 support PR, I suggest that we (automatically) use aarch64-apple-watchos-sim if the host is aarch64 and x86_64-apple-watchos-sim if the host is x86_64.

Perhaps we could extend the `auto-` mechanism to do this?

Reply:
I wonder what it will do if we use `-d sim` or '-d watchos-sim' actually. -d allows partial match. And if we are lucky the ios discovery code will only create an auto device for the matching architecture simulator...

And if it does not it may be desirable/easy to make it do so.